client/components/backup-storage-space/index.tsx

```diff
@@ -90,7 +91,7 @@ export const BackupStorageSpace: React.FC = () => {
 						upsellOption={ upsellOption }
 						usedStorage={ gigabytesUsed }
 						storageLimit={ gigabytesAvailable }
-						href="/pricing/backup"
+						href={ isJetpackCloud() ? '/pricing/backup' : '/plans' }
```

Comment:
<img width="1434" alt="Screen Shot 2021-08-04 at 11 49 17 AM" src="https://user-images.githubusercontent.com/2810519/128238399-bfcd065f-d50c-4b17-aa6e-8ac66fc7a296.png">

My local calypso is throwing a fit navigating directly to plans, but regardless we should be sure to stick the `siteSlug` here since otherwise it will go to the site selector?